### DISCUSSION TITLE
suggestion and autocompletion of magic (dunder) methods

### DISCUSSION THREAD

**Discussion by Badr-MOUFAD** [microsoft/pylance-release#2739] [Enhancement]
It would be nice to have suggestions for additional magic methods as well as their auto-completion.

For example, defining methods such as ``__len__``, ``__getitem__``, and ``__setitem__`` are useful when creating special data structures that behave as a list.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
